src/ui/CategoryFacet/CategoryFacet.ts

```diff
@@ -0,0 +1,48 @@
+import { Component } from '../Base/Component';
+import { IFieldOption, ComponentOptions } from '../Base/ComponentOptions';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { Dom, $$ } from '../../utils/Dom';
+import { Initialization } from '../Base/Initialization';
+import { exportGlobally } from '../../GlobalExports';
+import 'styling/_CategoryFacet';
+import { CategoryFacetTemplates } from './CategoryFacetTemplates';
+import { CategoryValueRoot } from './CategoryValueRoot';
+
+export interface CategoryFacetOptions {
+  field: IFieldOption;
+}
+
+export class CategoryFacet extends Component {
```

Comment:
I feel like this component is making it hard to interact with it using custom code.

All the "juicy" methods are hidden under a private property (categoryValueRoot).

Eg: Select value/un-select value/go back one level/ etc.

They should be exposed as first class member of the base component, and can be simple proxy for the real method in the sub-class.

Reply:
I agree. I tried exposing some of what I think would be useful but thought it would be easier to do a bit later when the component is closer to being done. Taking a note of this.